hydra-plutus/src/Hydra/Contract/HeadTokens.hs

```diff
@@ -49,25 +49,42 @@ validate initialValidator headValidator seedInput action context =
     Burn -> validateTokensBurning context
 {-# INLINEABLE validate #-}
 
+-- | When minting head tokens we want to make sure that:
+--
+-- * There is single state token that is paid into v_head, which ensures
+-- continuity. ('singleSTIsPaidToTheHead')
+--
+-- * The number of minted PTs == number of participants (+1 for the ST) evident
+-- from the datum. ('mintedPTsMatchParties')
+--
+-- * Ensure out-ref and the headId are in the datum of the first output of the
+-- transaction which mints tokens. ('seedInputIsConsumed')
```

Comment:
This is not what is actually done in `seedInputIsConsumed`. I would suggest we remove those references here if they are out-of-date already.

In fact, none of the checks are doing what is described: no where we are checking the datum at the head output.

Also: the spending seedInput is missing

Reply:
You are right, comments are not accurate at all.

Still, I find value in keeping them. 

What do you think if we fix them instead?

Reply:
On the other hand, what do you mean by: 

| Also: the spending seedInput is missing

Reply:
comments fixed by @ch1bo